src/FillArrays.jl

```diff
@@ -589,6 +590,30 @@ function in(x, A::RectDiagonal{<:Number})
     x == zero(eltype(A)) || x in A.diag
 end
 
+#########
+# mean, std
+#########
+
+mean(A::AbstractFill; dims=(:)) = mean(identity, A; dims=dims)
+function mean(f, A::AbstractFill; dims=(:))
+    val = float(f(getindex_value(A)))
+    dims isa Colon ? val : 
+        Fill(val, ntuple(d -> d in dims ? 1 : size(A,d), ndims(A)))
+end
+
+for fun in (:std, :var)
+    @eval function $fun(A::AbstractFill{T}; corrected::Bool=true, mean=nothing, dims=(:)) where {T<:Number}
+        if mean !== nothing
+            mean ≈ Statistics.mean(A; dims=dims) || throw(ArgumentError("pre-computed mean is incorrect"))
```

Comment:
Maybe this should change, the usual method does let you give the wrong mean -- I was lazy to make it precisely reproduce the same (wrong) answer. Perhaps just ignoring it would also be OK?
```
julia> std(fill(1,2), mean=2)
1.4142135623730951

julia> std(fill(1,10), mean=2)
1.0540925533894598
```